### PR TITLE
Improve moduleloader

### PR DESCRIPTION
**Pull request by danielkesselberg**
**Dangerous change:** Removed conditional caching from ModuleAutoLoader

### CODE REVIEW

File: protected/humhub/components/bootstrap/ModuleAutoLoader.php [hunk 1]
```diff
@@ -43,11 +62,52 @@ public function bootstrap($app)
...
+     * @return array
+     */
+    public static function findModulesByPath($path)
+    {
+        $hasConfigurationFile = function ($path) {
+            return is_file($path . DIRECTORY_SEPARATOR . self::CONFIGURATION_FILE);
+        };
+
+        try {
+            return FileHelper::findDirectories(Yii::getAlias($path, true), ['filter' => $hasConfigurationFile]);
```
Comment (githubjeka) [inline protected/humhub/components/bootstrap/ModuleAutoLoader.php]: https://www.yiiframework.com/doc/api/2.0/yii-helpers-basefilehelper#findDirectories()-detail

What about the case when

- moduleDir
   - tests
      - config.php
   - config.php

File: protected/humhub/components/bootstrap/ModuleAutoLoader.php [hunk 2]
```diff
@@ -43,11 +62,52 @@ public function bootstrap($app)
                     }
                 }
             }
-            if (!YII_DEBUG) {
```
Comment (githubjeka) [inline protected/humhub/components/bootstrap/ModuleAutoLoader.php]: Can you describe what's wrong with this line?

Comment (githubjeka) [reply protected/humhub/components/bootstrap/ModuleAutoLoader.php]: > Conditional caching could lead to unexpected results when YII_DEBUG is
enabled while cache is not DummyCache

Ah, ok

File: protected/humhub/components/bootstrap/ModuleAutoLoader.php (hunk 1 above)
Comment (danielkesselberg) [reply protected/humhub/components/bootstrap/ModuleAutoLoader.php]: Good catch. I will look into recursive true/false tomorrow.

File: protected/humhub/components/bootstrap/ModuleAutoLoader.php [hunk 3]
```diff
@@ -19,8 +20,26 @@
...
     public function bootstrap($app)
+    {
+        $modules = self::locateModules();
+
+        Yii::$app->moduleManager->registerBulk($modules);
+    }
+
+    /**
+     * Find available modules
+     * @deprecated 1.3 replace call for locateModules with findModules and handle caching outside of method (e.g. in boostrap)
```
Comment (githubjeka) [inline protected/humhub/components/bootstrap/ModuleAutoLoader.php]: How about remove `@deprecated` and rewrite it by `findModules()`, now.

**Comment by danielkesselberg** [comment]
@luke- Code Climate not happy but build is ok.

**Comment by acs-ferreira** [comment]
@danielkesselberg I believe Codeclimate can be fixed using `getOrSet()` here.

**Comment by danielkesselberg** [comment]
@acs-ferreira CodeClimate is unhappy because locateModules is way to complex. There is nothing we can do right now. I added `@depcreated` and added a cleaner version with some unit tests.

**Pull request merged**